code/game/machinery/status_display.dm

```diff
@@ -198,6 +169,44 @@
 	else
 		return "The display says:<br>\t<tt>Shuttle missing!</tt>"
 
+/**
+ * Nice overlay to make text smoothly scroll with no client updates after setup.
+ */
+/obj/effect/overlay/status_display_text
+	icon = 'icons/obj/status_display.dmi'
+	vis_flags = VIS_INHERIT_LAYER | VIS_INHERIT_PLANE | VIS_INHERIT_ID
+
+	var/message = ""
+
+/obj/effect/overlay/status_display_text/New(yoffset, line)
+	maptext_y = yoffset
+	message = line
+
+	var/line_length = length_char(line)
+
+	if(line_length > CHARS_PER_LINE)
+		// Marquee text
+		var/marquee_message = "[line] • [line] • [line]"
+		var/marqee_length = line_length * 3 + 6
+		maptext = generate_text(marquee_message, FALSE)
+		maptext_width = 6 * marqee_length
+		maptext_x = 32
+
+		// Mask off to fit in screen.
+		filters = filter(type = "alpha", icon = icon(icon, "outline"))
```

Comment:
We have a filter api, you should use it
![image](https://user-images.githubusercontent.com/58055496/153331331-384c4822-653b-4063-b006-1ff0a18ba674.png)

Reply:
done